src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1411,14 +1418,17 @@ static class MultiResolutionIconImage extends AbstractMultiResolutionImage {
         final Map<Integer, Image> resolutionVariants = new HashMap<>();
 
         public MultiResolutionIconImage(int baseSize, Map<Integer, Image> resolutionVariants) {
-            assert !resolutionVariants.containsValue(null)
-                   : "There are null icons in the MRI variants map";
+            if (resolutionVariants.containsValue(null)) {
+                throw new NullPointerException("There are null icons in the MRI variants map");
+            }
```

Comment:
I'm unsure about this. The idea with assert was to provide a debugging aid without affecting regular runtime. We control all the paths that call this constructor and, therefore, we should ensure it's never called if `resolutionVariants` contains `null`.

It's exactly what you've done in this PR.

Reply:
Ok, i will revert that part.